pandas/core/indexes/datetimes.py

```diff
@@ -603,7 +603,7 @@ def _wrap_setop_result(self, other, result):
             raise ValueError('Passed item and index have different timezone')
         return self._shallow_copy(result, name=name, freq=None, tz=self.tz)
 
-    def intersection(self, other):
+    def intersection(self, other, sort=True):
```

Comment:
The ``sort`` parameter is not fully implemented for ``intersection`` on ``DatetimeIndex`` yet. I just put this in for now to fix some failing tests